src/tools/cephfs/Resetter.cc

```diff
@@ -55,6 +63,13 @@ void Resetter::reset(int rank)
     if (r == -ENOENT) {
       cerr << "journal does not exist on-disk. Did you set a bad rank?"
```

Comment:
Refactored this to do the hard case only when "--hard" is passed, and point the user to that if primary reset fails.